metrics_api/lib/opentelemetry/metrics/instrument/asynchronous_instrument.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Metrics
+    module Instrument
+      # https://opentelemetry.io/docs/specs/otel/metrics/api/#asynchronous-instrument-api
+      class AsynchronousInstrument
+        attr_reader :name, :unit, :description, :callbacks
+
+        # @api private
+        def initialize(name, unit: nil, description: nil, callbacks: nil)
+          @name = name
+          @unit = unit || ''
+          @description = description || ''
+          @callbacks = callbacks ? Array(callbacks) : []
+        end
+
+        # @param callbacks [Proc, Array<Proc>]
```

Comment:
Technically, `callbacks` is an `Array<Proc>` now, always - I'm not sure how to document a "splat" param.